### PR TITLE
添加对 qBittorrentEE 的 ShadowBan API 的支持

### PR DESCRIPTION
**Pull request by Gaojianli**
实现比较粗糙，但是能用了
[TODO] 需要检查shadowban是否已经开启

## Summary by CodeRabbit

- **New Features**
	- Introduced a shadow banning option for more nuanced control over peer bans.
	- Enhanced ban management system with new configuration settings for shadow bans.
	- Added a new class for improved interaction with the qBittorrent API, including torrent management and peer bans.
	- Added support for a new downloader type labeled "qbittorrentee."
	- Added a user-friendly form component for configuring qBittorrent settings.
	- Improved user interface with new ShadowBan options in the dashboard and onboarding components.

- **Bug Fixes**
	- Adjusted logic for applying bans based on the new shadow ban functionality.

- **Documentation**
	- Updated configuration options to include shadow ban settings for user reference.
	- Added localization entries for ShadowBan functionality in English and Chinese.

### CODE REVIEW

**Review by Ghost-chu — CHANGES_REQUESTED** [review]
前端 WebUI 需要适配 qBittorrentEE 的OOBE、创建和编辑

**Comment by Ghost-chu** [comment]
推迟此 PR：qBittorrentEE 当前 shadow ban 实现有点问题，已反馈修复。

修复完成后，此 PR 除了要前端适配以外，还需要做 QBEE 的版本检查。

File: src/main/java/com/ghostchu/peerbanhelper/downloader/impl/qbittorrent/QBittorrentEE.java [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+                        , HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8));
+                if (request.statusCode() != 200) {
+                    log.error(tlUI(Lang.DOWNLOADER_QB_INCREAMENT_BAN_FAILED, name, apiEndpoint, request.statusCode(), "HTTP ERROR", request.body()));
+                    throw new IllegalStateException("Save qBittorrent banlist error: statusCode=" + request.statusCode());
+                }
+            } catch (Exception e) {
+                log.error(tlUI(Lang.DOWNLOADER_QB_INCREAMENT_BAN_FAILED, name, apiEndpoint, "N/A", e.getClass().getName(), e.getMessage()), e);
+                throw new IllegalStateException(e);
+            }
+        });
```
[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/java/com/ghostchu/peerbanhelper/downloader/impl/qbittorrent/QBittorrentEE.java]: Don't advice logger stuff anymore.

File: src/main/java/com/ghostchu/peerbanhelper/downloader/impl/qbittorrent/QBittorrentEE.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+    public boolean isLoggedIn() {
+        HttpResponse<Void> resp;
+        try {
+            resp = httpClient.send(MutableRequest.GET(apiEndpoint + "/app/version"), HttpResponse.BodyHandlers.discarding());
+        } catch (Exception e) {
+            return false;
+        }
+        return resp.statusCode() == 200;
+    }
```
[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/java/com/ghostchu/peerbanhelper/downloader/impl/qbittorrent/QBittorrentEE.java:142]: @coderabbitai Don't provide logging suggestions anymore.

**Comment by Ghost-chu** [comment]
移除 on-hold 状态，等待前端更改。

File: src/main/resources/lang/en_us/messages.yml [hunk 3]
```diff
@@ -365,5 +365,6 @@ IN_ECOMODE_SHORT: "🍃EcoQoS"
 IN_ECOMODE_DESCRIPTION: "EcoQoS API load successfully，Windows Efficiency Mode now applied to PeerBanHelper for saving power usage."
 
 DOWNLOADER_TRANSMISSION_DISCOURAGE: "Warning: Use Transmission Adapter is discourage. Frequent starting and stopping of torrents on seeds that are often subject to bans can result in frequent updates to the tracker server, indirectly triggering DoS attacks. This increases the load on the tracker server and may lead to your IP address being banned by the tracker. We encourage you to migrate to other downloaders whenever possible. https://github.com/PBH-BTN/PeerBanHelper/issues/382"
+DOWNLOADER_QBITTORRENTEE_SHADOWBANAPI_TEST_FAILURE: "You're using official qBittorrent build or the qBittorrentEE version you're using that not support ShadowBan, Please turn off ShadowBan switch."
```
[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/resources/lang/en_us/messages.yml:368]: @coderabbitai Make a commit for suggestion

**Review by Ghost-chu — APPROVED** [review]

**Review by paulzzh — APPROVED** [review]

**Pull request merged**